cpp/src/arrow/compute/light_array.cc

```diff
@@ -19,7 +19,9 @@
 
 #include <type_traits>
 
+#include "arrow/array/builder_binary.h"
```

Comment:
Well, there is a pitfall about `BaseBinaryBuilder::memory_limit()` - it's `std::numeric_limits<int32_t>::max() - 1`.  I only noticed this til my initial test failed. So I thought it might be necessary to keep using `BaseBinaryBuilder::memory_limit()` to avoid the potential mismatch. Does that make sense?

Reply:
Hmm, it might be an artifact of how `BinaryBuilder` is implemented, but it is not being used here.

Reply:
OK, I got the point. They are indeed not necessarily related. I'll update.

Reply:
Done.